astropy/utils/masked/function_helpers.py

```diff
@@ -1333,6 +1323,91 @@ def nanfunc(a, *args, **kwargs):
     )
 
 
+@dispatched_function
+def ediff1d(ary, to_end=None, to_begin=None):
+    from astropy.utils.masked import Masked
+
+    # ediff1d works fine if ary is Masked, but not if it is not (and
+    # we got here because to_end and/or to_begin are Masked).
+    if not isinstance(ary, Masked):
+        ary = Masked(ary)
+
+    return np.ediff1d.__wrapped__(ary, to_end, to_begin), None, None
+
+def _in1d(ar1, ar2, assume_unique=False, invert=False, *, kind=None):
+    # Copy sorting implementation from _arraysetops_impl._in1d;
+    # the others cannot work in the presence of a mask.
+    if kind == 'table':
+        raise ValueError(
+            "The 'table' method is not supported for Masked arrays."
+            "Please select 'sort' or None for kind."
+        )
+
+    # Straight copy from here on.
+    if not assume_unique:
+        ar1, rev_idx = np.unique(ar1, return_inverse=True)
+        ar2 = np.unique(ar2)
+
+    ar = np.concatenate((ar1, ar2))
+    # We need this to be a stable sort, so always use 'mergesort'
+    # here. The values from the first array should always come before
+    # the values from the second array.
+    order = ar.argsort(kind='mergesort')
```

Comment:
Do we need a way to let the user manage the sort?

Reply:
There is little choice - it has to be stable sort (this is copied directly from the numpy implementation).